core-primitives/types/src/parentchain.rs

```diff
@@ -40,3 +41,15 @@ pub type BlockHash = sp_core::H256;
 
 /// Alias to 512-bit hash when used in the context of a transaction signature on the chain.
 pub type Signature = MultiSignature;
+
+#[derive(Encode, Decode, Copy, Clone, Debug, PartialEq, Eq)]
+pub enum ParentchainId {
+	/// Primary parentchain containing the teerex pallet.
+	Teerex,
```

Comment:
not only teerex, also sidechain finality.
I suggest we simply name this "Integritee"